### PR TITLE
Web launch url allow https

### LINKED ISSUE #109369: `flutter run --web-launch-url` should allow URL to be https
I have been working on getting `flutter run -d chrome` to sit behind a reverse proxy so I can debug with an environment more closely matching production such as  ssl, Secure headers, CSP, COOP and COEP.   One issue among many to making this work I discovered by accident.  

Once the `flutter run` command launches the browser and opens the default http://localhost:xxx  url any other browser tabs that attempt to connect via the reverse proxy are unable to fully load,  I don't pretend to understand why, some interaction with sse or wss.   Whatever it is I'm over trying to figure it out... maybe.   

What does work is never launching on the **http** url and instead opening the proxied **https** url. However the `--web-launch-url` flag doesn't allow an https url like `https://localhost:9921`

Looks like the problem is with the regex here not allowing https.

https://github.com/flutter/flutter/blob/47f54ace4568c079a2fdb66260d61a74ef5cca03/packages/flutter_tools/lib/src/web/web_device.dart#L134

### PR DESCRIPTION
**Pull request by nullrocket**
Allow https URL scheme in addition to plain http when using  `flutter run --web-launch-url`

```bash
# currently only allows http like 
flutter run --web-launch-url=http://localhost:1234

# this PR will allow https also
flutter run --web-launch-url=https://localhost:1234
```
**Note:** This doesn't affect what is served by flutter run, that is still plain http.  It just allows launching a different url in chrome than the one being served by flutter run 

Fixes #109369

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/web/web_device.dart [hunk 1]
```diff
@@ -131,7 +131,7 @@ abstract class ChromiumDevice extends Device {
     // for the web initialization and server logic.
     String url;
     if (debuggingOptions.webLaunchUrl != null) {
-      final RegExp pattern = RegExp(r'^((http)?:\/\/)[^\s]+');
+      final RegExp pattern = RegExp(r'^((http|https)?:\/\/)[^\s]+');
```
Comment (LongXiangGuo) [inline packages/flutter_tools/lib/src/web/web_device.dart:134]:
```suggestion
      final RegExp pattern = RegExp(r'^(https?:\/\/)[^\s]+');
```

File: packages/flutter_tools/lib/src/web/web_device.dart [hunk 2]
```diff
@@ -131,7 +131,7 @@ abstract class ChromiumDevice extends Device {
     // for the web initialization and server logic.
     String url;
     if (debuggingOptions.webLaunchUrl != null) {
-      final RegExp pattern = RegExp(r'^((http)?:\/\/)[^\s]+');
+      final RegExp pattern = RegExp(r'^(https?:\/\/)[^\s]+');
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/web/web_device.dart:134]: can you add a comment explaining the tool still only serves http?

File: packages/flutter_tools/test/commands.shard/hermetic/run_test.dart [hunk 3]
```diff
@@ -876,7 +876,7 @@ void main() {
     final DebuggingOptions options = await command.createDebuggingOptions(true);
     expect(options.webLaunchUrl, 'http://flutter.dev');
 
-    final RegExp pattern = RegExp(r'^((http)?:\/\/)[^\s]+');
+    final RegExp pattern = RegExp(r'^((http|https)?:\/\/)[^\s]+');
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/hermetic/run_test.dart:879]: Can you add a test that passes a `web-launch-url` with https that passes the regex test?

**Comment by christopherfujino** [comment]
Thank you for contributing! As you have not responded within 3 weeks, I will be closing this PR as stale. If you are still interested in landing this, feel free to rebase this PR upstream, address code review feedback, and re-open this PR.

**Pull request closed without merging**